WalletWasabi.Gui/ViewModels/ViewModelBase.cs

```diff
@@ -1,50 +1,53 @@
-using Newtonsoft.Json;
 using ReactiveUI;
 using System;
 using System.Collections;
-using System.Collections.Generic;
 using System.ComponentModel;
-using System.Linq;
-using System.Reflection;
-using WalletWasabi.Gui.ViewModels.Validation;
-using WalletWasabi.Models;
+using WalletWasabi.Gui.Validation;
 
 namespace WalletWasabi.Gui.ViewModels
 {
-	public class ViewModelBase : ReactiveObject, INotifyDataErrorInfo
+	public class ViewModelBase : ReactiveObject, INotifyDataErrorInfo, IRegisterValidationMethod
 	{
+		private Validations _validations;
+
 		public ViewModelBase()
 		{
-			var vmc = Validator.PropertiesWithValidation(this).ToList();
-			if (vmc.Count == 0)
-			{
-				return;
-			}
-
-			ValidationMethodCache = vmc;
+			_validations = new Validations();
+			_validations.ErrorsChanged += OnValidations_ErrorsChanged;
+			PropertyChanged += ViewModelBase_PropertyChanged;
 		}
 
 		public event EventHandler<DataErrorsChangedEventArgs> ErrorsChanged;
 
-		private List<(string propertyName, MethodInfo mInfo)> ValidationMethodCache { get; }
-
-		public bool HasErrors => Validator.ValidateAllProperties(this, ValidationMethodCache).HasErrors;
+		protected IValidations Validations => _validations;
 
-		public IEnumerable GetErrors(string propertyName)
+		private void OnValidations_ErrorsChanged(object sender, DataErrorsChangedEventArgs e)
 		{
-			var error = Validator.ValidateProperty(this, propertyName, ValidationMethodCache);
+			ErrorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(e.PropertyName));
+		}
 
-			if (error.HasErrors)
+		private void ViewModelBase_PropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (string.IsNullOrWhiteSpace(e.PropertyName))
+			{
+				_validations.Validate();
+			}
+			else
 			{
-				return error;
+				_validations.ValidateProperty(e.PropertyName);
 			}
+		}
+
+		bool INotifyDataErrorInfo.HasErrors => (_validations as IValidations).Any;
```

Comment:
This property should be moved above the methods.

Reply:
Instead of `(_validations as IValidations)` you can use `Validations`.

Reply:
done